src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextViewConnectionListener.cs

```diff
@@ -63,11 +121,131 @@ public void SubjectBuffersConnected(ITextView textView, ConnectionReason reason,
             }
 
             RazorLSPTextViewFilter.CreateAndRegister(vsTextView);
+
+            lock (_lock)
+            {
+                _activeTextViews.Add(textView);
+
+                // Initialize the user's options and start listening for changes.
+                // We only want to attach the option changed event once so we don't receive multiple
+                // notifications if there is more than one TextView active.
+                if (!textView.TextBuffer.Properties.ContainsProperty(RazorEditorTrackedView) &&
+                    textView.TextBuffer.IsRazorLSPBuffer())
+                {
+                    // We assume there is ever only one TextBuffer at a time and thus all active
+                    // TextViews have the same TextBuffer.
+                    _textBuffer = textView.TextBuffer;
+
+                    var bufferOptions = _editorOptionsFactory.GetOptions(_textBuffer);
+                    _textBuffer.Properties[RazorEditorBufferOptions] = bufferOptions;
+
+                    // All TextViews share the same options, so we only need to listen to changes for one.
+                    _textBuffer.Properties[RazorEditorTrackedView] = textView;
+
+                    var razorEditorTextViewOptions = _editorOptionsFactory.GetOptions(textView);
```

Comment:
Could you elaborate on why we need both the textview options here and the buffer ones above? Also, given that you only track one textview at a time, and assuming it's needed, it'll be important to have a good comment explaining why we only ever update options on one and not all

Reply:
Added these comments to the code:
```
// We need to keep track of and update both the TextView and TextBuffer options. Updating
// the TextView's options is necessary so 'SPC'/'TABS' in the bottom right corner of the
// view displays the right setting. Updating the TextBuffer is necessary since it's where
// LSP pulls settings from when sending us requests.
```

```
// A change in Tools->Options settings only kicks off an options changed event in the view
// and not the buffer, i.e. even if we listened for TextBuffer option changes, we would never
// be notified. As a workaround, we listen purely for TextView changes, and update the
// TextBuffer options in the TextView listener as well.
```
^this latter issue was discovered via testing. I'm not sure why a Tools->Options change doesn't kick off a TextBuffer option changed event even if we listen for it, but I think this workaround works.

Reply:
@gundermanc lol VS' details make me sad